src/components/Article.jsx

```diff
@@ -0,0 +1,132 @@
+import { Link, useNavigate } from "react-router-dom";
+import {
+  Card,
+  CardMedia,
+  CardContent,
+  Typography,
+  Button,
+  CardActions,
+  Box,
+} from "@mui/material";
+import { fetchArticleById } from "../utils/api";
+import { useEffect, useState } from "react";
+
+export default function Article({ article }) {
+  const [articleWithBody, setArticleWithoBody] = useState(null);
+
+  useEffect(() => {
+    fetchArticleById(article.article_id).then((response) => {
+      setArticleWithoBody(response);
+    });
+  }, [article.article_id]);
+
```

Comment:
There is no need to fetch the article by id for an article card. All you need to do is map over the articles you have already fetched. 

This api call is for when you want to go to a separate article page :)